addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -436,6 +436,11 @@ async function loadImageInfo(img, rpc, attachmentSrc = '') {
     // user domains could have been changed meanwhile.
     let relativeSrc;
     try {
+        if (src.startsWith("//")) {
+            // The url is a protocol-relative url. Transform it into an absolute
+            // url first.
+            src = `${window.location.protocol}${src}`;
```

Comment:
As a general principle, take URL construction as a red flag if it's not using the development languages provided tools or lib with use.

In this case, this should definitely go through `new URL()`

```suggestion
            src = new URL(src, window.location.href).href;
```

Seems to do exactly the same from my testing.
Does it change anything then? Probably not.
Does it matter then? Yes, we are sure to not have a bug due to wrong concat at some point, and if we have, it will be through `new URL()` clear error raising.

Reply:
Good point, good thing I did not merge it like that yesterday because I was about too after our discord discussion to change your commit message and some more. Actually what was done here might not have been needed at all: https://github.com/odoo/odoo/commit/89c14783846288a2de53f6258a93440e02550b13

-> Always use one call to new URL with the right second parameter, always take the pathName, and we won't have need the `[?#]` split afterwards.

To be even better, probably do not use `window` directly but `img.ownerDocument.defaultView` (something like that)

Reply:
> Good point, good thing I did not merge it like that yesterday because I was about too after our discord discussion to change your commit message and some more.

I guess there is discussion I am not aware of, should I let you handle this one @qsm-odoo ?

Reply:
Yes I am handling it, thanks :+1: